lib/screens/map_view.dart

```diff
@@ -52,64 +52,75 @@ class _MapViewScreenState extends State<MapViewScreen> {
     }
-    locationData = await locationService.getLocation();
     setState(() {});
   }
 
   Widget map(BuildContext context) {
+    MapControllerImpl mapController = MapControllerImpl();
     if (locationData == null) {
       return const Center(child: CircularProgressIndicator());
     } else {
-      return FlutterMap(
-        options: MapOptions(
-          center: latlng.LatLng(
-              locationData?.latitude ?? 51.5, locationData?.longitude ?? -0.09),
-          zoom: 18.0,
-          interactiveFlags: InteractiveFlag.none,
-        ),
-        layers: [
-          TileLayerOptions(
-              urlTemplate: mapboxURL, // ignore: undefined_identifier
-              attributionBuilder: (_) {
-                return const Text("© Mapbox");
-              },
-              additionalOptions: {
-                'accessToken': mapboxAPIKey, // ignore: undefined_identifier
-                'id': 'mapbox.mapbox-streets-v8'
-              }),
-          MarkerLayerOptions(
-            markers: [
-              Marker(
-                  width: 40.0,
-                  height: 40.0,
-                  point: latlng.LatLng(locationData?.latitude ?? 51.5,
-                      locationData?.longitude ?? -0.09),
-                  builder: (ctx) => const Beastie())
-            ],
-          ),
-        ],
-      );
+      return StreamBuilder(
+          stream: locationService.onLocationChanged.asBroadcastStream(),
+          builder: (context, AsyncSnapshot<LocationData> currLocation) {
+            if (!currLocation.hasData) {
+              return const Center(child: CircularProgressIndicator());
+            }
+            latlng.LatLng mapCenter = latlng.LatLng(
+                currLocation.data?.latitude ?? 51.5,
+                currLocation.data?.longitude ?? -0.09);
+            const zoomLevel = 18.0;
+            mapController.onReady.then((_) {
+              mapController.move(mapCenter, zoomLevel);
+            });
+            return FlutterMap(
+              mapController: mapController,
+              options: MapOptions(
+                zoom: zoomLevel,
+                interactiveFlags: InteractiveFlag.none,
+              ),
+              layers: [
+                TileLayerOptions(
+                    urlTemplate: mapboxURL, // ignore: undefined_identifier
+                    attributionBuilder: (_) {
+                      return const Text("© Mapbox");
+                    },
+                    additionalOptions: {
+                      'accessToken':
+                          mapboxAPIKey, // ignore: undefined_identifier
+                      'id': 'mapbox.mapbox-streets-v8'
+                    }),
+                MarkerLayerOptions(
+                  markers: [
+                    Marker(
+                        width: 40.0,
+                        height: 40.0,
+                        point: latlng.LatLng(
+                            currLocation.data?.latitude ?? 51.5,
```

Comment:
A side effect of this change is that now the marker here will also move along with the user. But I imagine this marker is still here just for testing so it's probably fine for now